Feature/222284 lingo re branding and UI update

samridhis opened the pull request:
latest UI changes

sppanday committed:
moved liberate your communicate outside setting

sppanday committed:
aligned logo and color as per requirement

Commit:
rebranding changes

Commit:
Merge branch 'feature/221259-rebranding-changes2' of https://github.com/fhirfactory/pegacorn-communicate-matrix-react-sdk into feature/221259-rebranding-changes

# Conflicts:
#	src/components/structures/HomePage.tsx

Commit:
Home page redesign

Commit:
Tabbed view act logo fix

Commit:
Login page rebranding UI changes

jrschremmer commented on res/css/views/auth/_AuthHeader.scss:
```diff
@@ -20,6 +20,7 @@ limitations under the License.
     width: 206px;
     padding: 25px 25px;
     box-sizing: border-box;
+    background-color: white!important;
```
Hi Samridhi,  please review how we have been implementing branding (e.g. https://dev.azure.com/ACTGovernment/Pegacorn/_workitems/edit/219811).  As we want to be able to push our fork back to element at some stage (to minimise merge issues when we pull from element master when required) we can't change the default behaviour of the code.  Instead either implement the change via:
1. The "Pegacorn communicate - Light theme custom colors" custom theme
2. Controlled by a config.json configuration property that defaults to the current value in the element code, if the propoerty is not in config.json
3. We have our own copy of the file in aether-host-files/common/element-web/matrix-react-sdk (this is the least favoured solution as it doesn't address the merge issues with element master code, but has been done for a quick tactical solution)

samridhis replied:
![image](https://user-images.githubusercontent.com/12698292/116013333-0cbb8d80-a673-11eb-9f67-cb02c610dddc.png)
I just went through changes from Surendra on the changeset. I am able to see his changes on the same file.

jrschremmer replied:
Yes but that file is in a different git repository, which is option 3 I described above.  If you implement your changes in that repo and revert the changes in this repo, that will be fine.

samridhis replied:
Thanks, I have reverted this specific change as it was already a part of aether host files.

jrschremmer commented on src/components/views/auth/PasswordLogin.tsx:
```diff
@@ -484,7 +484,7 @@ export default class PasswordLogin extends React.PureComponent<IProps, IState> {
                     { !this.props.busy && <input className="mx_Login_submit"
                         type="submit"
-                        value={_t('Sign in')}
+                        value={('Log in')}
```
This breaks internationalisation in element.  I'd suggest updating the value in src\i18n\strings\en_EN.json

jrschremmer commented on src/components/views/elements/SSOButtons.tsx:
```diff
@@ -58,7 +58,7 @@ const SSOButton: React.FC<ISSOButtonProps> = ({
     ...props
 }) => {
-    const label = idp ? _t("Continue with %(provider)s", { provider: idp.name }) : _t("Sign in with single sign-on");
+    const label = idp ? _t("Continue with %(provider)s", { provider: idp.name }) : ("Log in with single sign-on");
```
This breaks internationalisation in element. I'd suggest updating the value in src\i18n\strings\en_EN.json

samridhis committed:
Updated string name Signin to Login on the login page

jrschremmer commented on src/components/structures/HomePage.tsx:
```diff
@@ -119,10 +119,7 @@ const HomePage: React.FC<IProps> = ({ justRegistered = false }) => {
 
         introSection = <React.Fragment>
-            <img src={logoUrl} alt={config.brand} />
-            {showLiberateYourCommunicationText &&
```
Why has the "showLiberateYourCommunicationText &&" block of code been removed?  If we don't want to show this text, it can be controlled by configuration as implemented correctly by Surendra.

samridhis replied:
Done

jrschremmer commented on src/components/structures/HomePage.tsx:
```diff
@@ -119,14 +119,7 @@ const HomePage: React.FC<IProps> = ({ justRegistered = false }) => {
-            {customConfig.showPrimaryLogoInAuthenticatedScreen &&
-            <img src={logoUrl} alt={config.brand}/>
-            }
-            {showWelcomeToElementText ?
```
Why has the "showWelcomeToElementText" block of code been removed? If we don't want to show this text, it can be controlled by configuration as implemented correctly by Surendra.

samridhis replied:
Done

jrschremmer commented on src/components/structures/TabbedView.tsx:
```diff
@@ -125,6 +125,14 @@ export default class TabbedView extends React.Component<IProps, IState> {
                     {labels}
                 </div>
                 {panel}
+                <div className="mx_AuthFooter_TabbedView">
```
Please conditionally show this block of code based on a configuration property (my suggestion would be  AuthFooterTabbedViewImgPath) having a value, that defaults to null, so the standard element behaviour isn't changed.  We can set the configuration property to themes/element/img/logos/ACTGov_inline_rev_black.png to show it for us.  The alt text needs to also be externalised into the configuration file.

samridhis replied:
Done

jrschremmer commented on src/components/views/auth/AuthHeader.js:
```diff
@@ -29,7 +29,12 @@ export default class AuthHeader extends React.Component {
         return (
-            <div className="mx_AuthHeader">
+            <div className="mx_AuthHeader"> 
+                <br/>  
```
Please conditionally show this block of code based on a configuration property (my suggestion would be AuthHeaderImgPath) having a value, that defaults to null, so the standard element behaviour isn't changed. We can set the configuration property to themes/element/img/logos/ACT-Government-inline-back-trans.jpg to show it for us.  The alt text needs to also be externalised into the configuration file.

samridhis replied:
Done

Commit:
change was arleady part of aether host file so reverting it back from the shared source.

Commit:
updated the configuration for Log in instead of Sign in.

samridhis committed:
tabbed view changes display using condition in configration

samridhis committed:
Merge branch 'feature/222284-Lingo-Re-Branding-and-UI-Update' of https://github.com/fhirfactory/pegacorn-communicate-matrix-react-sdk into feature/222284-Lingo-Re-Branding-and-UI-Update

samridhis committed:
reverting back deletion of configuration changes.

samridhis committed:
code review comment to display act logo on login page at the auth header .

jrschremmer commented on src/components/views/auth/AuthHeader.js:
```diff
@@ -27,13 +28,16 @@ export default class AuthHeader extends React.Component {
         const AuthHeaderLogo = sdk.getComponent('auth.AuthHeaderLogo');
         const LanguageSelector = sdk.getComponent('views.auth.LanguageSelector');
+        const loginScreen = SdkConfig.get().loginScreen;
+        const showAuthHeaderLogo_Act = loginScreen?.displayActLogo ?? true;
```
Please remove Act from the config property name and the variable name.  I think showAuthHeaderLogo should default to false e.g. ?? false, so the default behaviour of element is not affected.  Also please be consistent with the current code and centralise this into src/config.ts

samridhis replied:
done

jrschremmer commented on src/components/structures/TabbedView.tsx:
```diff
@@ -118,13 +119,24 @@ export default class TabbedView extends React.Component<IProps, IState> {
                     {labels}
                 </div>
                 {panel}
+                {customConfig.tabbedViewDisplayActLogo === true &&
```
You haven't externalised themes/element/img/logos/ACTGov_inline_rev_black.png and the alt text into the configuration file, as discussed in the previous conversation (I was unable to reply in that conversation in the github UI).  Also please remove Act from the config name.  Also please add the description of any new config values to feature-communicate-list.md

samridhis replied:
done

jrschremmer commented on res/css/structures/_HomePage.scss:
```diff
@@ -33,7 +33,7 @@ limitations under the License.
 
     img {
-        height: 48px;
+        height: 120px;
```
As mentioned in another conversation, please implement all changes as either:

1. Controlled by a config.json configuration property that defaults to the current value in the element code, if the propoerty is not in config.json
2. We have our own copy of the file in aether-host-files/common/element-web/matrix-react-sdk (this is the least favoured solution as it doesn't address the merge issues with element master code, but has been done for a quick tactical solution)

samridhis replied:
done

jrschremmer commented on res/css/views/auth/_AuthFooter.scss:
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 
 .mx_AuthFooter {
-    text-align: center;
+    text-align: right;
```
As mentioned in another conversation, please implement all changes as either:
1. Controlled by a config.json configuration property that defaults to the current value in the element code, if the propoerty is not in config.json
2. We have our own copy of the file in aether-host-files/common/element-web/matrix-react-sdk (this is the least favoured solution as it doesn't address the merge issues with element master code, but has been done for a quick tactical solution)

samridhis replied:
done

jrschremmer replied:
Hi Sam, this change is still hard-coded in the code.  Please externalise it into the theme like $font-14px is.

samridhis replied:
sorry done now

jrschremmer commented on res/css/views/auth/_AuthHeaderLogo.scss:
```diff
@@ -21,11 +21,30 @@ limitations under the License.
 
 .mx_AuthHeaderLogo img {
-    width: 100%;
+    //width: 100%;
```
As mentioned in another conversation, please implement all changes as either:
1. Controlled by a config.json configuration property that defaults to the current value in the element code, if the propoerty is not in config.json
2. We have our own copy of the file in aether-host-files/common/element-web/matrix-react-sdk (this is the least favoured solution as it doesn't address the merge issues with element master code, but has been done for a quick tactical solution)

samridhis replied:
Current style is set as below in this style width is repeacted twice. I am not sure what is the benifit of doing that ? 
.mx_AuthHeaderLogo img {
    width: 100%;
    width: 150px;
    height: 100px;
}

samridhis replied:
done

jrschremmer commented on res/css/views/auth/_AuthPage.scss:
```diff
@@ -19,7 +19,7 @@ limitations under the License.
     display: flex;
     flex-direction: column;
-    background-color: $authpage-bg-color;
+    background-color: #002677!important;
```
I suggest you change the value of $authpage-bg-color in our copy of themes\light\css\_light.scss in aether-host-files and avoid !important, as it should only be used as a last resort.

jrschremmer replied:
This comment is still unresolved

samridhis replied:
done

jrschremmer commented on res/css/structures/_TabbedView.scss:
```diff
@@ -27,7 +27,9 @@ limitations under the License.
 }
-
+.mx_AuthFooter_Image_Tabbed{
+height: 120px!important;
```
Is !important required here ... it should only be used as a last resort.  NOTE: this change doesn't have to move to aether-host-files as it is adding new code and not making any updates.

samridhis replied:
Done

jrschremmer commented on res/themes/light/css/_light.scss:
```diff
@@ -14,7 +14,7 @@ $monospace-font-family: Inconsolata, Twemoji, 'Apple Color Emoji', 'Segoe UI Emo
 // unified palette
 // try to use these colors when possible
-$accent-color: #0DBD8B;
+$accent-color:  #002677;
```
Please move this change to aether-host-files

samridhis replied:
done

jrschremmer commented on src/components/views/auth/AuthHeader.js:
```diff
@@ -27,9 +28,16 @@ export default class AuthHeader extends React.Component {
+                {showAuthHeaderLogo_Act &&                  
+                <div>
+                    <br/>              
+                    <img className="mx_AuthHeaderLogo_Act" src="themes/element/img/logos/ACT-Government-inline-back-trans.jpg" alt="ACT Govt Logo"  />                
```
As mentioned in another comment, please externalised the img src path and the alt text into the configuration file and remove any references to "Act"

samridhis replied:
Done

Commit:
replace act with secondary in regards to logo

Commit:
removed !imortant tag as it was not needed

Commit:
moved landing page logo height style into config file.

Commit:
reverted back removal of width.

Commit:
latest configuration changes for branding UI changes discussed on 03rd may 2021

jrschremmer commented on res/css/views/auth/_AuthHeaderLogo.scss:
```diff
@@ -22,10 +22,29 @@ limitations under the License.
 
 .mx_AuthHeaderLogo img {
     width: 100%;
+    width: 150px;
```
There are 2 definitions for width here, please externalise a single value into the themes.  The values in the element themes will be 100% and our copy of the light theme in aether-host-files can specify 150px.  For the height value the value in the element themes, you will have to work out as based on https://developer.mozilla.org/en-US/docs/Web/CSS/height I'm not sure if previously having no value is equivalent to inherit, unset or auto.  In our copy of the light theme the value would be 100px

samridhis replied:
to avoid confusion i have added this file in aether host files with the changes..

jrschremmer commented on res/css/views/elements/_SSOButtons.scss:
```diff
@@ -46,13 +46,13 @@ limitations under the License.
-        color: $button-primary-bg-color;
-        background-color: $button-secondary-bg-color;
-        border-color: $button-primary-bg-color;
+        color: $button-secondary-bg-color;
```
These changes look odd, as the default button is being assigned the secondary colours which is not the expected behaviour.

samridhis replied:
this change needed for the light color button now instead of the white button previously for SSO

jrschremmer replied:
But you have changed the default element behaviour, which we need to avoid ... changing the value in our custom theme of $button-primary-bg-color to the value of $button-secondary-bg-color maybe a solution but that could have other unintended consequences ... perhaps based on config change the style name used e.g. instead of using .mx_SSOButton_default use .mx_SSOButton_default_inverse which you will need to add.

samridhis replied:
done

jrschremmer commented on res/css/views/elements/_SSOButtons.scss:
```diff
@@ -46,13 +46,13 @@ limitations under the License.
     .mx_SSOButton_default.mx_SSOButton_primary {
-        color: $button-primary-fg-color;
-        background-color: $button-primary-bg-color;
+        color: $button-primary-bg-color;
```
This change looks odd as the foreground and background colours have been swapped?

samridhis replied:
it's all good

samridhis replied:
this change needed for light color button now instead of white button previously for SSO

jrschremmer replied:
Same as the other comment on the file.

samridhis replied:
done

jrschremmer commented on res/themes/light/css/_light.scss:
```diff
@@ -14,7 +14,7 @@ $monospace-font-family: Inconsolata, Twemoji, 'Apple Color Emoji', 'Segoe UI Emo
 // unified palette
 // try to use these colors when possible
-$accent-color: #0DBD8B;
+$accent-color:  #00818b;
```
Changes to this file need to be reverted, so we don't change the default behaviour of element.  Instead these changes go into our copy of this file in aether-host-files

samridhis replied:
done

jrschremmer commented on src/components/structures/TabbedView.tsx:
```diff
@@ -118,13 +119,24 @@ export default class TabbedView extends React.Component<IProps, IState> {
+                        <a href="" target="_blank" rel="noreferrer noopener">                        
+                        </a>
+                        <span className="mx_AuthFooter_brand_TabbedView">
+                            <img className="mx_AuthFooter_Image_Tabbed" src="themes/element/img/logos/ACTGov_inline_rev_black.png" alt="ACT Govt Logo"/>
```
The image path and ACT Govt Logo need to be externalised into the config too pls

jrschremmer commented on src/components/views/auth/PasswordLogin.tsx:
```diff
@@ -433,7 +433,8 @@ export default class PasswordLogin extends React.PureComponent<IProps, IState> {
 
         let loginType;
-        if (!SdkConfig.get().disable_3pid_login) {
+        const disable_3pid_login =SdkConfig.get().loginScreen?.disable_3pid_login??false;
```
Please revert this change and continue to use the element default config location for disable_3pid_login

samridhis replied:
I am not sure why to revert this change. there are multiple configurations inside the login screen which i have kept into the login screen config. it will be easier for a person to identify the configuration by that.
"loginScreen": {
"displaySecondaryLogo":false,
"displaySecondaryVerticalLogo":false,
"displayLanguageSelector":false,
"disable_3pid_login": false,
"showHomeServerInfo": false,
"showMatrixDefaultFooterLinks": false,
"allowPhoneBasedAuthentication": false,
"allowEmailBasedAuthentication": false,
"changeSigninToLoginTextLabel": true,
"backgroundColor": "(0,38, 119)",
"logo":{
"img":{
"width": "150px",
"height": "100px"
}
}

jrschremmer replied:
Because you are changing the default element behaviour, which we are trying to avoid

samridhis replied:
done

jrschremmer commented on src/components/views/auth/RegistrationForm.tsx:
```diff
@@ -411,7 +411,7 @@ export default class RegistrationForm extends React.PureComponent<IProps, IState
 
     private showPhoneNumber() {
-        const threePidLogin = !SdkConfig.get().disable_3pid_login;
+        const threePidLogin = !SdkConfig.get().loginScreen?.disable_3pid_login;
```
Please revert this change and continue to use the element default config location for disable_3pid_login

samridhis replied:
@jrschremmer I am not sure why to revert this change. there are multiple configurations inside the login screen which i have kept into the login screen config. it will be easier for a person to identify the configuration by that.
 "loginScreen": {
        "displaySecondaryLogo":false,
        "displaySecondaryVerticalLogo":false,        
        "displayLanguageSelector":false,    
        "disable_3pid_login": false,            
        "showHomeServerInfo": false,
        "showMatrixDefaultFooterLinks": false,
        "allowPhoneBasedAuthentication": false,
        "allowEmailBasedAuthentication": false,
        "changeSigninToLoginTextLabel": true,
        "backgroundColor": "(0,38, 119)",
        "logo":{
                "img":{
                "width": "150px",
                "height": "100px"
            }
        }

samridhis replied:
done

jrschremmer commented on src/config.ts:
```diff
@@ -20,7 +25,7 @@ export const signInTextNeedsToBeReplaced = loginScreen?.changeSigninToLoginTextL
 // background color on login screen
 // used by app-web to load a colored background instead of lake image
-export const loginScreenBackgroundColor = loginScreen?.backgroundColor ?? null;
+export const showLoginScreenBackgroundImage = loginScreen?.showLoginScreenBackgroundImage ?? null;
```
Searching this pull request, I can't see where showLoginScreenBackgroundImage is still used?

samridhis replied:
done

jrschremmer commented on src/config.ts:
```diff
@@ -43,3 +48,34 @@ export const showHomeServerDetail = loginScreen?.showHomeServerInfo ?? true;
+// show primary logo on logged in screen
+// Due to positioning and etc this switch would make it easy to have that logo configured.
+// it is to control logo that element provides
+export const showPrimaryLogoOnLoginScreen = logoConfig?.showPrimaryLogoOnLoginScreen ?? true;
```
I can't see where the showPrimaryLogo* config items are still used?  Please review all the config items in this file and any still used please ensure their description is added to feature-communicate-list.md

samridhis replied:
done

samridhis committed:
removed unused variables from config.ts file

samridhis committed:
tabbed view footer image url and alt text passing onto configuration file

samridhis committed:
revert back the color change.

samridhis committed:
revert back text-align changes

samridhis committed:
code review comments fix

samridhis committed:
$accent-color changed back to default and moved to aether-host-files

Commit:
pushing the changes

Commit:
removed unused variable

Commit:
Merge branch 'feature/222284-Lingo-Re-Branding-and-UI-Update' of https://github.com/fhirfactory/pegacorn-communicate-matrix-react-sdk into feature/222284-Lingo-Re-Branding-and-UI-Update

# Conflicts:
#	res/themes/light/css/_light.scss
#	src/config.ts

Commit:
externalise authheader alignment.

Commit:
removed newly added variable as its moved to common

jrschremmer commented on src/config.ts:
```diff
@@ -43,3 +46,19 @@ export const showHomeServerDetail = loginScreen?.showHomeServerInfo ?? true;
+// tab condition
+export const tabbedView =config['tabbedView'];
+export const tabbedViewdisplaySecondaryLogo  = tabbedView?.displaySecondaryLogo ?? false;
+export const logoUrlTabbedViewFooter =  tabbedView?.imgUrl;
```
For consistency please rename 
imgUrl to secondaryLogoUrl
description to secondaryLogoAltText

samridhis replied:
done

Commit:
$authfooter-text-align:alighnment changes

samridhis committed:
disable_3pid_login fix

jrschremmer committed:
CODE REVIEW of 222284: Revert changes to files where no functional change remained (e.g. _common.scss, AuthHeaderLogo.js, LanguageSelector.js);
Revert change to PasswordLogin.tsx, as not required, as this code is hidden if the room server only supports SSO login, so we control this behaviour via that method instead;
In the themes move the definition of $authfooter-text-align:center to be grouped with the other $auth variables, which is then just above $dark-panel-bg-color in all themes;
In _light.scss move the new mx_TopBanner styles up from the bottom of the file to avoid merge conflicts with new lines added to element;

jrschremmer commented on src/components/structures/TabbedView.tsx:
```diff
@@ -118,13 +119,23 @@ export default class TabbedView extends React.Component<IProps, IState> {
                 {panel}
+                {customConfig.tabbedViewdisplaySecondaryLogo === true &&
+                    <div className="mx_AuthFooter_TabbedView">
+                        <a href="" target="_blank" rel="noreferrer noopener">                        
```
What is this empty hyperlink for?  If it is no longer required pls remove it

samridhis replied:
Sure

samridhis replied:
done

samridhis committed:
as per review comments

samridhis committed:
documentation update for newly added configuration changes.

samridhis committed:
initial fixes

samridhis committed:
revert back wrong push meant for different story

jrschremmer committed:
CODE REVIEW: 222284-Lingo-Rebranding - Consolidate the different previous solution used for branding, in particular move the styling configuration out of config.json and into /res/themes/light/css/_light.scss, as some of the styling in config.json was being overwritten by /res/themes/light/css/_light.scss;
Remove config for secondary logos that was no longer required, to simplify testing. If require again the implementation can be referenced via the check in history;
Rollback changes to the following files as they are no longer required: src/i18n/strings/en_EN.json, src/i18n/strings/en_US.json, src/components/views/auth/PasswordLogin.tsx;
Fix the Login Screen impl, so it works for both use cases of the room server having password authentication enabled and not enabled;
Add the background gradient to the Login Screen;
Fix the implementation in _TabbedView.scss so the bottom div for the secondary logo, didn't impede on the right hand panel;
Change config starting with display* to show*, to be consistent with previous config names;

jrschremmer committed:
Fix whitespace

jrschremmer committed:
Re-added loginScreen.showHomeServerInfo;
